lib/src/core/verifier/mod.rs

```diff
@@ -0,0 +1,76 @@
+use crate::*;
+use anyhow::Result;
+use async_trait::async_trait;
+use serde_json::{json, Value};
+use std::path::PathBuf;
+
+pub mod policy;
+mod sample;
+mod sgx;
+mod tdx;
+
+#[derive(Debug)]
+pub struct Evidence {
+    pub quote: String,
+    pub ehd: String,
+    pub add: String,
+}
+
+#[derive(Debug)]
+pub struct AttestationResults {
+    pub tee: String,
+    pub attestation_type: String,
+    pub policy: String,
+    pub add: String,
+}
+
+#[async_trait]
+pub trait TEEtraits {
+    fn parse(&self, evidence: &Value) -> Result<Evidence>;
+    async fn evaluate(&self, evidence: &Evidence) -> Result<AttestationResults>;
+    fn opa_policy_path(&self) -> Result<PathBuf>;
+    fn opa_reference_path(&self) -> Result<PathBuf>;
+}
+
+#[derive(Debug)]
+pub struct Verifiers {
+    pub tdx: tdx::Tdx,
+    pub sgx: sgx::Sgx,
+    pub sample: sample::Sample,
+}
+
+impl Verifiers {
+    pub async fn new(workdir: PathBuf) -> Result<Self> {
+        Ok(Self {
+            tdx: tdx::Tdx::new(workdir.clone()).await?,
+            sgx: sgx::Sgx::new(workdir.clone()).await?,
+            sample: sample::Sample::new(workdir.clone()).await?,
+        })
+    }
+}
+
+#[async_trait]
+impl Verifier for Verifiers {
+    async fn evaluate(&self, evidence: &Value) -> Result<String> {
+        let tee = evidence["tee"]
+            .as_str()
+            .ok_or(anyhow!("convert tee error"))?;
+        let results = match tee {
+            "sample" => Ok(self.sample.evaluate(&self.sample.parse(evidence)?).await?),
+            "sgx" => Ok(self.sgx.evaluate(&self.sgx.parse(evidence)?).await?),
+            "tdx" => Ok(self.tdx.evaluate(&self.tdx.parse(evidence)?).await?),
+            _ => Err(anyhow!("not supported!")),
+        }?;
+        attestation_results(&results)
+    }
+}
+
+fn attestation_results(results: &AttestationResults) -> Result<String> {
```

Comment:
It might be more idiomatic to make an `AttestationResults` struct that is returned by the evaluate method of each verifier. Then you could just serialize the struct here.